### PR TITLE
Grade week plan

### PR DESCRIPTION
**Pull request by Stoldt**
## Description

Changes to API and various useHooks. Added provider

## Type of PR

- Feature
- Refactor

## Related Issues/Tickets

Please link the related issues/tickets here.

## Checklist

- [X] Completed the user story described by the issue (if applicable)
- [X] Code follows the code guidelines
- [X] Tests have been added/updated (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: __tests__/useActivity.test.tsx [hunk 1]
```diff
@@ -6,7 +6,8 @@ import useActivity, {
   FullActivityDTO,
   useSingleActivity,
 } from "../hooks/useActivity";
-import CitizenProvider from "../providers/CitizenProvider";
+import WeekplanProvider from "../providers/WeekplanProvider";
+import { createActivityCitizen } from "../apis/activityAPI";
```
Comment (PixieStick314) [inline __tests__/useActivity.test.tsx:10]: Check warning on line 10 in __tests__/useActivity.test.tsx
'createActivityCitizen' is defined but never used

File: app/auth/profile/organisation/grade/[grade].tsx [hunk 2]
```diff
@@ -7,11 +7,13 @@ import IconButton from "../../../../../components/IconButton";
 import useGrades, { GradeDTO } from "../../../../../hooks/useGrades";
 import { colors, ScaleSize, ScaleSizeH, ScaleSizeW, SharedStyles } from "../../../../../utils/SharedStyles";
 import SearchBar from "../../../../../components/SearchBar";
+import { useWeekplan } from "../../../../../providers/WeekplanProvider";
 
 const ViewGrade = () => {
+  const [searchedCitizens, setSearchedCitizens] = useState<string>("");
+  const { setIsCitizen, setId, id } = useWeekplan();
```
Comment (PixieStick314) [inline app/auth/profile/organisation/grade/[grade].tsx:14]: Check warning on line 14 in app/auth/profile/organisation/grade/[grade].
'id' is assigned a value but never used

File: hooks/useActivity.ts [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
-  fetchByDateRequestForCitizen,
+  fetchByDateCitizen,
+  fetchByDateGrade,
   toggleActivityStatusRequest,
   updateRequest,
 } from "../apis/activityAPI";
+import { useWeekplan } from "../providers/WeekplanProvider";
 
-import { useCitizen } from "../providers/CitizenProvider";
+/* It's highly recommended to read the docs at https://tanstack.com/query/latest
```
Comment (PixieStick314) [inline hooks/useActivity.ts:15]: What's this for?

**Review by PixieStick314 — COMMENTED** [review]
Der er lige nogle ting der skal kigges på, github klager i hvert fald.

**Review by MertzAndreas — APPROVED** [review]
Im impressed, good German engineering ❤️ 🙋🏿‍♂️ 🇩🇪 🇩🇪 

Ich bin beeindruckt, gute deutsche Ingenieurskunst ❤️ 🙋🏿‍♂️ 🇩🇪 🇩🇪 

tumblr_mj2ansGhYP1qesvk3o1_500

File: hooks/useActivity.ts (hunk 3 above)
Comment (MertzAndreas) [reply hooks/useActivity.ts:15]: Caching library that we use, hard to document everything in the codebase. So we just linked the docs

**Review by Tbjerregaard — APPROVED** [review]
LEONGERMAN

**Pull request merged**